sdk/cosmosdb/cosmos/README.md

```diff
@@ -1,50 +1,230 @@
 const client = new CosmosClient({ endpoint, key });
 
-const databaseDefinition = { id: "sample database" };
-const collectionDefinition = { id: "sample collection" };
-const documentDefinition = { id: "hello world doc", content: "Hello World!" };
+async function main() {
+  // The rest of the README samples are designed to be pasted into this function body
+}
+
+main().catch((error) => {
+  console.error(error);
+});
+```
+
+For simplicity we have included the `key` and `endpoint` directly in the code but you will likely want to load these from a file not in source control using a project such as [dotenv](https://www.npmjs.com/package/dotenv)
+
+In production environments, secrets like keys should be stored in [Azure Key Vault](https://azure.microsoft.com/en-us/services/key-vault/)
+
+## Key concepts
+
+Once you've initialized a [CosmosClient](https://docs.microsoft.com/en-us/javascript/api/@azure/cosmos/cosmosclient?view=azure-node-lates), you can interact with the primary resource types in Cosmos DB:
+
+- [Database](https://docs.microsoft.com/en-us/javascript/api/@azure/cosmos/database?view=azure-node-latest): A Cosmos DB account can contain multiple databases. When you create a database, you specify the API you'd like to use when interacting with its documents: SQL, MongoDB, Gremlin, Cassandra, or Azure Table. Use the [Database](https://docs.microsoft.com/en-us/javascript/api/@azure/cosmos/database?view=azure-node-latest) object to manage its containers.
+
+- [Container](https://docs.microsoft.com/en-us/javascript/api/@azure/cosmos/container?view=azure-node-latest): A container is a collection of JSON documents. You create (insert), read, update, and delete items in a container by using methods on the [Container](https://docs.microsoft.com/en-us/javascript/api/@azure/cosmos/container?view=azure-node-latest) object.
+
+- [Item](https://docs.microsoft.com/en-us/javascript/api/@azure/cosmos/item?view=azure-node-latest): An Item is a JSON document stored in a container. Each Item you add to a container must include an `id` key with a value that uniquely identifies the item within the container.
+
+For more information about these resources, see [Working with Azure Cosmos databases, containers and items][cosmos_resources].
+
+## Examples
+
+The following sections provide several code snippets covering some of the most common Cosmos DB tasks, including:
+
+- [Create a database](#create-a-database)
+- [Create a container](#create-a-container)
+- [Insert items](#insert-items)
+- [Query documents](#query-the-database)
+- [Read an item](#read-an-item)
+- [Delete an item](#delete-an-data)
+
+### Create a database
+
+After authenticating your [CosmosClient](https://docs.microsoft.com/en-us/javascript/api/@azure/cosmos/cosmosclient?view=azure-node-latest), you can work with any resource in the account. The code snippet below creates a SQL API database.
+
+```js
+const { database } = await client.databases.createIfNotExists({ id: "Test Database" });
```

Comment:
I don't quite understand your point here. They are different. And service level concepts. Do you have a suggestion for how else to talk about them?

Reply:
No - I mean "Test Database" vs "testDatabase" or "testdatabase". Having a space in the actual name of the database / container feels wrong to me. Like those should be one-word things, but I can't think of a valid reason why that would be other than it might effect the SQL syntax? But I'm not sure it does.

Reply:
I understand now. Makes sense.